File: shopshare/src/main/java/com/sp5blue/shopshare/dtos/listitem/ListItemDto.java
package com.sp5blue.shopshare.dtos.listitem;

import com.sp5blue.shopshare.models.listitem.ListItem;

import java.time.LocalDateTime;
import java.util.UUID;

public record ListItemDto(String id, String name, String status, UUID createdBy, Boolean locked, LocalDateTime createdOn) {
    public ListItemDto(ListItem listItem) {
        this(
                listItem.getId().toString(),
                listItem.getName(),
                listItem.getStatus().name(),
                listItem.getCreatedBy().getId(),
                listItem.isLocked(),
                listItem.getCreatedOn()
        );
    }
}

File: front-end/models/listitem/ListItemDto.ts
import ListItemStatus from './ListItemStatus';

// String id, String name, String status, UUID createdBy, Boolean locked, LocalDateTime createdOn
type ListItemDto = {
  id: string;
  name: string;
  status: ListItemStatus;
  createdBy: string;
  locked: boolean;
  createdOn: string;
};

export default ListItemDto;
